StreamingUpdatesRestApi/Program.cs

```diff
@@ -367,6 +380,45 @@ public static async Task<bool> MainAsync(bool test = false)
 
                     Console.WriteLine();
                     #endregion
+
+                    // Step 13
+                    // Create additional signups and make an API request to GetAllSignups with query parameters to view all signups
+                    #region Step13
+                    Console.WriteLine("Step 13: Get All Signups");
+
+                    // Create additional signups
+                    for (int i = 0; i < AdditionalSignupsToCreate; i++)
+                    {
+                        // The signup does not need to be active to view in Get All Signups. 
+                        CreateSignupInput additionalSignupToCreate = new CreateSignupInput()
+                        {
+                            Name = $"{SignupName}_{i}",
+                            ResourceType = ResourceType.Stream,
+                            ResourceIds = streamsToAdd,
+                        };
+
+                        using StringContent additionalSignupToCreateString = new (JsonSerializer.Serialize(additionalSignupToCreate));
+                        additionalSignupToCreateString.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+                        response = await httpClient.PostAsync(new Uri($"{resource}/api/{apiVersion}/Tenants/{tenantId}/Namespaces/{namespaceId}/signups", UriKind.Absolute), additionalSignupToCreateString).ConfigureAwait(false);
+                        CheckIfResponseWasSuccessful(response);
+                    }
+
+                    response = await httpClient.GetAsync(new Uri($"{resource}/api/{apiVersion}/Tenants/{tenantId}/Namespaces/{namespaceId}/signups?skip={GetAllSignupsSkip}&count={GetAllSignupsCount}", UriKind.Absolute)).ConfigureAwait(false);
+                    CheckIfResponseWasSuccessful(response);
+
+                    SignupsWrapper signupsWrapper = JsonSerializer.Deserialize<SignupsWrapper>(await response.Content.ReadAsStreamAsync().ConfigureAwait(false), _apiJsonOptions);
```

Comment:
I'm not sure how I feel about having a signup wrapper class for deserialization. Would specifying the type as ```IEnumerable<Signup>``` or ```List<Signup>``` not be sufficient?

Reply:
We need a wrapper class because the model changed to have the signups wrapped in a container for the response. I have changed the name of the class to SignupCollection to be consistent with our codebase. This class is similar to the SignupResources class.

Here's an example of the response for Get All Signups
```
{
	"signups": [
		{
			"id": "c6ee04b2-da7f-4c9a-94f5-bc7ee9b7fc2d",
			"name": "Signup_1",
			"owner": {
				"type": "Client",
				"objectId": "f3944a6f-9c25-4b4c-ab0f-35b5b8d7e6af",
				"tenantId": "a52749df-e223-44d7-b33c-09ef85ba73bb"
			},
			"communityId": "",
			"type": "Stream",
			"createdDate": "2023-11-21T18:15:17+00:00",
			"modifiedDate": "2023-11-21T18:15:17+00:00",
			"signupState": "Activating"
		}
	]
}
```

Reply:
Gotcha. I am surprised the response is not just a list of signups, but I'm sure there is a reason for that.